README.md

```diff
@@ -0,0 +1,79 @@
+# khealth
+=====
+
+[![Docker Image on Quay.io](https://quay.io/repository/coreos/khealth/status "Docker Image on Quay.io")](https://quay.io/repository/coreos/khealth)
+
+*khealth*  is a Kubernetes cluster monitoring suite. Its Routines exercise Kubernetes subsystems and send events to Collectors. Collectors collate these events to compute current cluster state. Cluster status is available from Collectors over a simple HTTP API, which is served on a cluster nodeport in the example below.
+
+## Quick start
+
+If you have a kubernetes cluster, you can deploy khealth.
+
+```sh
+cd khealth/
+kubectl create -f ./contrib/k8s/khealth-ns.yaml
+kubectl create -f ./contrib/k8s/khealth-{rc,service}.yaml
+```
+
+This will create a nodeport service which exposes the following status endpoints.
+
+| Command  | NodePort |
+| ------------- |:-------------:|
+| rcscheduler  | 31337 |
+
+## Architecture
+
+A khealth *Module* is a single command that invokes a set of Routines and a single Collector. The Collector gathers events from the Routines and exposes metrics on its status endpoint.
+
+* **cmd/**:
+
+  This is where the `Module` entrypoint programs live. Each `Module` should have exactly one `main` package package in an eponymous directory beneath `cmd/`.
+
+* **pkg/routines**:
+
+  Routines are defined in structures that implement the `RoutineHandler` interface.
+  ```go
+  type RoutineHandler interface {
+  	Init() error
+	Poll() error
+	Cleanup() error
+  }
+  ```
+  `Init` is called, and then `Poll` in a loop. Once the ttl has expired, `Cleanup` is called. Rinse-repeat. Every step generates an event which is received by the collector.
+
+  The `NewRoutine` function can be used to create a new Routine.
```

Comment:
The `NewRoutine` function [takes x, y, z where x/y/z may be args(?) and] returns a [pointer to?] a new khealth Routine.